src/window/WindowPostMessageStream.ts

```diff
@@ -11,6 +12,26 @@ interface WindowPostMessageStreamArgs {
   targetWindow?: Window;
 }
 
+const getSourceDescriptor = Object.getOwnPropertyDescriptor(
+  MessageEvent.prototype,
+  'source',
+);
+assert(
+  getSourceDescriptor,
+  'MessageEvent.prototype.source getter is not defined.',
+);
+const getSource = getSourceDescriptor.get;
+
+const getOriginDescriptor = Object.getOwnPropertyDescriptor(
+  MessageEvent.prototype,
+  'origin',
+);
+assert(
+  getOriginDescriptor,
+  'MessageEvent.prototype.origin getter is not defined.',
+);
+const getOrigin = getOriginDescriptor.get;
```

Comment:
This ensures that both getters are defined, and removes the need for the other assert calls in `_onMessage`.

```suggestion
const getSource = Object.getOwnPropertyDescriptor(
  MessageEvent.prototype,
  'source',
)?.get;
assert(
  getSource,
  'MessageEvent.prototype.source getter is not defined.',
);

const getOrigin = Object.getOwnPropertyDescriptor(
  MessageEvent.prototype,
  'origin',
)?.get;
assert(
  getOrigin,
  'MessageEvent.prototype.origin getter is not defined.',
);
```

Reply:
I tried that, but it fails to build:
![Screenshot 2023-03-06 at 10 23 40](https://user-images.githubusercontent.com/13301024/223069437-c2f1c51d-d1a2-4e35-acee-8a61905dbd08.png)

Reply:
Do the functions have to be declared in the global scope, or can it be in the function scope too?

Reply:
Yes, they have to be in the global scope.

Reply:
How about this:

```ts
const getSource = Object.getOwnPropertyDescriptor(
  MessageEvent.prototype,
  'source',
)?.get;

const getOrigin = Object.getOwnPropertyDescriptor(
  MessageEvent.prototype,
  'origin',
)?.get;

assert(getSource, 'MessageEvent.prototype.source getter is not defined.');
assert(getOrigin, 'MessageEvent.prototype.origin getter is not defined.');

export class WindowPostMessageStream {
  // [...]

  private _onMessage(/* ... */) {
    /* eslint-disable @typescript-eslint/no-non-null-assertion */
    if (
      (this._targetOrigin !== '*' &&
        getOrigin!.call(event) !== this._targetOrigin) ||
      getSource!.call(event) !== this._targetWindow ||
      !isValidStreamMessage(message) ||
      message.target !== this._name
    ) {
      return;
    }
    /* eslint-enable @typescript-eslint/no-non-null-assertion */

    this._onData(message.data);
  }

  // [...]
}

Reply:
Why do they have to be in the global scope?

Reply:
I also tried the proposed solution with `/* eslint-disable @typescript-eslint/no-non-null-assertion */` and `!`, but it becomes a problem with Jest and coverage. I tried to design some test, but it seems that none of the approaches work with Jest since this is at the global level.
![Screenshot 2023-03-06 at 11 13 13](https://user-images.githubusercontent.com/13301024/223090885-ea39a160-f78f-484a-951e-9494adda45eb.png)

In my opinion I would prefer keeping the changes on the PR as they're now since no `!` or `// ignore-expect` comments are added which solves the initial request. If you think this still can be better and there are strong reasons for changing it, let me know I will try again your ideas.

Reply:
I've added the proposed solution with some additional changes. The only way I found is to ignore these two lines in terms of code coverage. I tried covering it with several approaches which include changing this directly or applying mock functions, none of it worked, jest execution environment would always ignore it somehow.

Let me know if this way is acceptable and preferred, I can agree to keep it this way.